src/components/v5/common/Fields/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,180 @@
+        setEndDate(end);
+      } else {
+        setStartDate(dates);
+      }
+    },
+    [selectsRange],
+  );
+
+  const resetValues = useCallback(() => {
+    if (selectsRange) {
+      setStartDate(startDateProp || null);
+      setEndDate(endDateProp || null);
+    } else {
+      setStartDate(selectedProp || null);
+      setEndDate(null);
+    }
+  }, [endDateProp, selectedProp, selectsRange, startDateProp]);
+
+  const {
+    onClick: applyButtonOnClick,
+    mode: applyButtonMode,
+    text: applyButtonText,
+  } = applyButtonProps || {};
+  const {
+    onClick: cancelButtonOnClick,
+    mode: cancelButtonMode,
+    text: cancelButtonText,
+  } = cancelButtonProps || {};
+
+  return (
+    <DatePicker<never, WithRange>
+      portalId={isMobile ? DATEPICKER_PORTAL_ID : undefined}
+      calendarContainer={DatepickerContainer}
+      calendarClassName={styles.wrapper}
+      renderCustomHeader={(props) => (
+        <DatepickerCustomHeader
+          startDate={startDate}
+          endDate={selectsRange ? endDate : null}
+          dateFormat={dateFormat}
+          setStartDate={setStartDate}
+          setEndDate={selectsRange ? setEndDate : undefined}
+          onClose={
+            isMobile
+              ? () => {
+                  resetValues();
+                  calendarRef.current?.setOpen(false);
+                }
+              : undefined
+          }
+          {...props}
+        />
+      )}
+      ref={calendarRef}
+      customInput={
+        selectsRange ? (
+          <DatepickerCustomRangeInput
+            startDate={startDateProp}
+            endDate={endDateProp}
+            dateFormat={dateFormat}
+          />
+        ) : (
+          <DatepickerCustomInput />
+        )
+      }
+      dateFormat={dateFormat}
+      popperClassName={clsx(popperClassName, 'max-w-[20.5rem] !z-[1000]')}
+      renderDayContents={(day) => (
+        <div className="react-datepicker__day-content">{day}</div>
+      )}
+      shouldCloseOnSelect={false}
+      popperModifiers={[
+        ...(popperModifiers || []),
+        {
+          name: 'offset',
+          options: {
+            offset: [10, -38],
+          },
+        },
+      ]}
+      selectsRange={selectsRange}
+      onChange={handleChange}
+      selected={selectsRange ? undefined : startDate}
+      startDate={selectsRange ? startDate : undefined}
+      endDate={selectsRange ? endDate : undefined}
+      onClickOutside={resetValues}
+      {...rest}
+    >
+      {((selectsRange && startDate && endDate) ||
+        (!selectsRange && startDate)) && (
+        <div className="p-4 border-t border-t-gray-200 w-full flex items-center justify-between gap-2">
+          <Button
+            {...cancelButtonProps}
+            mode={cancelButtonMode || ('primaryOutline' as ButtonMode)}
+            onClick={(e) => {
+              resetValues();
+              calendarRef.current?.setOpen(false);
+              cancelButtonOnClick?.(e);
+            }}
```

Comment:
Please rename this variable to variable to `event` and add the proper synthetic event type _(I believe it's `MouseEvent`)_